From 10b5057b5aea690254a8baa6e4672e3b2c3b64ab Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Fri, 26 Dec 2014 18:22:42 +0800
Subject: [PATCH 189/196] atomisp2: [WORKAROUND]Support crop region in
 set_parameters

Now, the hal only set crop region in per_frame setting. but
the crop region need to effect all config pipe.
the patch set crop region to all config pipe when set per_frame
setting to css.

Change-Id: I5fe7788f4f1215febb6285a12787da92cce7b22e
Tracked-On: 1503901631
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/34325
Reviewed-by: Tian Shu Qiu <tian.shu.qiu@intel.com>
Reviewed-by: Jing Tao <jing.tao@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Approver: Wei Tao <wei.tao@intel.com>
Reviewed-by: Wei Tao <wei.tao@intel.com>
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_cmd.c        |  3 +--
 .../media/pci/atomisp2/atomisp_driver/atomisp_fops.c       | 14 ++++++++++++++
 2 files changed, 15 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 8469409..817c472 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -2778,8 +2778,7 @@ void atomisp_apply_css_parameters(
 	if (arg->dp_config)
 		atomisp_css_set_dp_config(asd, &css_param->dp_config);
 
-	if (arg->dz_config &&
-		atomisp_check_zoom_region(asd, arg) == true)
+	if (arg->dz_config && (asd->run_mode->val != ATOMISP_RUN_MODE_VIDEO))
 		atomisp_css_set_dz_config(asd, &css_param->dz_config);
 
 	if (arg->nr_config)
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
index 7da5134..31238cc 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_fops.c
@@ -130,6 +130,20 @@ int atomisp_q_video_buffers_to_css(struct atomisp_sub_device *asd,
 						vm_mem->vaddr);
 			atomisp_css_update_isp_params_on_pipe(asd,
 				asd->stream_env[stream_id].pipes[css_pipe_id]);
+			/*
+			 * WORKAROUND:
+			 * Because the camera halv3 can't ensure to set zoom
+			 * region to per_frame setting and global setting at
+			 * same time and only set zoom region to pre_frame
+			 * setting now.so when the pre_frame setting inculde
+			 * zoom region,I will set it to global setting.
+			 */
+			if (param->us_params.dz_config &&
+				asd->run_mode->val != ATOMISP_RUN_MODE_VIDEO) {
+				atomisp_css_set_dz_config(asd,
+						&param->params.dz_config);
+				atomisp_css_update_isp_params(asd);
+			}
 			/* free the parameters */
 			atomisp_free_css_parameters(&param->params);
 			atomisp_kernel_free(param);
-- 
1.9.1

